File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = lambda: sys.stdin.readline().strip()

n, k = map(int, input().split())
arr = list(map(int, input().split()))
answer = []

left = 0
right = 0
summ = arr[right]

while left <= right:


    if summ < k:
        right += 1
        if right == n:
            break
        summ += arr[right]
    else:
        answer.append(right - left + 1)
        summ -= arr[left]
        left += 1

if answer:
    print(min(answer))
else:
    print(0)

File: 백준/Gold/1806. 부분합/부분합.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            while (!st.hasMoreTokens()) st = new StringTokenizer(br.readLine());
            arr[i] = Integer.parseInt(st.nextToken());
        }

        if (n == 0) {
            System.out.println(0);
            return;
        }

        int left = 0, right = 0;
        long sum = arr[right];             
        int minLen = Integer.MAX_VALUE;

        while (left <= right) {
            if (sum < k) {
                right++;
                if (right == n) break;     
                sum += arr[right];
            } else {
                minLen = Math.min(minLen, right - left + 1);
                sum -= arr[left];
                left++;
            }
        }

        System.out.println(minLen == Integer.MAX_VALUE ? 0 : minLen);
    }
}
